apiserver/plane/app/views/external/base.py

```
@@ -3,69 +1,169 @@
+    # Validate model is supported by provider
+    if model not in provider.models:
+        log_exception(ValueError(
+            f"Model {model} not supported by {provider.name}. "
+            f"Supported models: {', '.join(provider.models)}"
+        ))
+        return None, None, None
+
+    return api_key, model, provider_key
+
+
+def get_llm_response(task, prompt, api_key: str, model: str, provider: str) -> Tuple[str | None, str | None]:
```

Ruff (0.8.2)

Line too long (109 > 88)

(E501)